topics/release-notes.md

```diff
@@ -0,0 +1,1653 @@
+  metrics. ([#674])
+* Reduce redundant calls to `prepareClientToWrite` for continuous `addReply*`.
+  ([#670])
+* Optimize the logic for checking conversion to skip list during `ZADD` operations.
+  ([#806])
+* Optimize `sdsfree` with `zfree_with_size` to avoid redundant size calculation.
+  ([#453])
+* Combine events to eliminate redundant `kevent(2)` calls. ([#638])
+* Introduce shared query buffer for client reads to reduce memory usage. ([#258])
+* Optimize CRC64 performance for large batches by processing bytes in parallel.
+  ([#350])
+* Use `SOCK_NONBLOCK` to reduce system calls for outgoing connections. ([#293])
+* Enable `accept4()` detection on specific versions of various platforms. ([#294])
+* Convert CRC16 slot table to fixed-size array for improved memory efficiency.
+  (Redis#13112)
+* Run `SCRIPT FLUSH` truly asynchronously and close Lua interpreter in a
+  background thread. (Redis#13087)
+* Optimize `DEL` command to avoid redundant deletions for expired keys. (Redis#13080)
+* Improve defragmentation for large bins to enhance memory efficiency. (Redis#12996)
+* Optimize hash table resizing to include empty dictionaries. (Redis#12819)
+* Reduce performance impact of dictionary rehashing by optimizing bucket processing.
+  (Redis#12899)
+* Optimize performance for simultaneous client `[P|S]UNSUBSCRIBE`. (Redis#12838)
+* Optimize CPU cache efficiency during dictionary rehashing. (Redis#5692)
+* Optimize `ZRANGE` offset location from linear search to skip list jump. (Redis#12450)
+* Fix `aeSetDontWait` timing to avoid unnecessary waits in `aeProcessEvent`. (Redis#12068)
+
+#### <a id="800-rc1-performanceefficiency-improvements-cluster"></a>Performance/Efficiency Improvements - Cluster
+
+* Add lightweight cluster message header for Pub/Sub messages. ([#654])
+* Minor performance improvement in Valkey cluster by avoid initializing key
+  buffer in `getKeysResult`. ([#631])
+* Cache `CLUSTER SLOTS` response to improve throughput and reduce latency. ([#53])
+* Replace slots_to_channels radix tree with slot-specific dictionaries for
+  shard channels. (Redis#12804)
+* Optimize `KEYS` command when pattern includes hashtag and implies a single
+  slot. (Redis#12754)
+* Optimize `SCAN` command with `MATCH` when pattern implies a single slot.
+  (Redis#12536)
+* Replace cluster metadata with slot specific dictionaries to reduce memory
+  usage when using Valkey cluster. (Redis#11695, Redis#12704)
+
+#### <a id="800-rc1-reliability-improvements-core"></a>Reliability Improvements - Core
+
+* Limit tracking custom errors (e.g. from Lua) while allowing normal errors
+  to be tracked ([#500], Redis#13141)
+* Manage maximum number of new connections per cycle to prevent connection
+  storms. (Redis#12178)
+
+#### <a id="800-rc1-reliability-improvements-cluster"></a>Reliability Improvements - Cluster
+
+* Reduce fail-over time in Valkey cluster when multiple sequential fail-overs
```

Comment:
```suggestion
* Reduce failover time in Valkey cluster when multiple sequential fail-overs
```

Reply:
The sentence seems incomplete. Can we amend it a bit? when multiple sequential failovers -> during multiple sequential failovers?